server/src/main/java/io/druid/segment/loading/LocalDataSegmentKiller.java

```diff
@@ -35,50 +36,36 @@
   @Override
   public void kill(DataSegment segment) throws SegmentLoadingException
   {
-    final File path = getDirectory(segment);
-    log.info("segment[%s] maps to path[%s]", segment.getIdentifier(), path);
+    final File path = getPath(segment);
+    log.info("killing segment[%s] mapped to path[%s]", segment.getIdentifier(), path);
 
-    if (!path.isDirectory()) {
-      if (!path.delete()) {
-        log.error("Unable to delete file[%s].", path);
-        throw new SegmentLoadingException("Couldn't kill segment[%s]", segment.getIdentifier());
-      }
-
-      return;
-    }
+    try {
+      if (path.getName().endsWith(".zip")) {
 
-    final File[] files = path.listFiles();
-    int success = 0;
+        // path format -- > .../dataSource/interval/version/partitionNum/xxx.zip
+        File partitionNumDir = path.getParentFile();
+        FileUtils.deleteDirectory(partitionNumDir);
 
-    for (File file : files) {
-      if (!file.delete()) {
-        log.error("Unable to delete file[%s].", file);
+        //try to delete other directories if possible
+        File versionDir = partitionNumDir.getParentFile();
+        if (versionDir.delete()) {
+          File intervalDir = versionDir.getParentFile();
+          if (intervalDir.delete()) {
+            File dataSourceDir = intervalDir.getParentFile();
+            dataSourceDir.delete();
+          }
+        }
       } else {
-        ++success;
+        throw new SegmentLoadingException("Unknown file type[%s]", path);
       }
     }
-
-    if (success == 0 && files.length != 0) {
-      throw new SegmentLoadingException("Couldn't kill segment[%s]", segment.getIdentifier());
-    }
-
-    if (success < files.length) {
-      log.warn("Couldn't completely kill segment[%s]", segment.getIdentifier());
-    } else if (!path.delete()) {
-      log.warn("Unable to delete directory[%s].", path);
-      log.warn("Couldn't completely kill segment[%s]", segment.getIdentifier());
+    catch (IOException e) {
+      throw new SegmentLoadingException(e, "Unable to kill segment");
     }
   }
 
-  private File getDirectory(DataSegment segment) throws SegmentLoadingException
+  private File getPath(DataSegment segment) throws SegmentLoadingException
   {
-    final Map<String, Object> loadSpec = segment.getLoadSpec();
-    final File path = new File(MapUtils.getString(loadSpec, "path"));
-
-    if (!path.exists()) {
-      throw new SegmentLoadingException("Asked to load path[%s], but it doesn't exist.", path);
-    }
-
-    return path.getParentFile();
+    return new File(MapUtils.getString(segment.getLoadSpec(), "path"));
```

Comment:
Is there any way to tie this to LoadLocalSpec ? parsing the loadspec and calling getPath seems a little extreme for here, but can "path" be put as a `public static final String` in the class or something?